### PR TITLE
[dy] Support base path configuration

### PR DESCRIPTION
**Pull request by dy46**
# Summary

Add support for `MAGE_BASE_PATH` environment variable. This PR is to support adding a prefix to a Mage URL, i.e. (localhost:6789/my_prefix/...)

To support this I added some additional logic on top of https://github.com/mage-ai/mage-ai/pull/426. I created an additional next config file (next_base_path.config.js) that will be used in the export frontend CI job to create a new `frontend_dist_base_path` folder. Once the server is started and the base path env var is set, we will try to replace the pre-configured base path string to the base path that the user has set.

# Tests

tested locally

Screenshot 2023-07-31 at 4 04 15 PM
Screenshot 2023-07-31 at 4 03 23 PM

cc: @wangxiaoyou1993 @johnson-mage

### CODE REVIEW

**Comment by wangxiaoyou1993** [comment]
Could you share more testing details (e.g. screenshots)?

**Review by johnson-mage — APPROVED** [review]
Went through the code diff and didn't see any issues, though the main assurance will be from testing the app with some base path config

File: mage_ai/server/server.py [hunk 1]
```diff
@@ -196,11 +226,22 @@ def make_app():
         (r'/templates', MainPageHandler),
         (r'/version-control', MainPageHandler),
     ]
+
+    if update_routes:
```
Comment (wangxiaoyou1993) [inline mage_ai/server/server.py:230]: Could you add unit tests to the update_routes logic?

File: mage_ai/server/server.py [hunk 2]
```diff
@@ -106,7 +110,30 @@ def post(self, action_type):
         self.write(dict(scheduler=dict(status=scheduler_manager.get_status())))
 
 
-def make_app():
+def replace_base_path(base_path: str) -> None:
```
Comment (wangxiaoyou1993) [inline mage_ai/server/server.py:113]: Could you add unit tests to this method as well?

**Comment by dy46** [comment]
@wangxiaoyou1993 addressed your comments

**Comment by wangxiaoyou1993** [comment]
You can rebase master branch. The linter errors were fixed on master branch.

**Comment by dy46** [comment]
@wangxiaoyou1993 I think the linting for my PR might be a little messed up because I squashed some commits. It might think I edited some files that I didn't end up editing

**Review by wangxiaoyou1993 — APPROVED** [review]

**Pull request merged**